### ISSUE TITLE
A floating point exception can be triggered in DenseToCSRSparseMatrix

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#58179]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.11.0-dev20221013

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.8

### CUDA/cuDNN version

CUDA 11.5

### Current Behaviour?

```shell
A floating point exception can be triggered in DenseToCSRSparseMatrix, which leads to a crash.
```

### Standalone code to reproduce the issue

```shell
import os
os.environ['TF_ENABLE_ONEDNN_OPTS'] = '0'
import tensorflow as tf
import numpy as np
print(tf.__version__)
for _ in range(20):
    try:
        with tf.device("GPU:0"):
            dense_input = tf.complex(tf.random.uniform([0, 1, 1], dtype=tf.float32),tf.random.uniform([0, 1, 1], dtype=tf.float32))
            indices = tf.saturate_cast(tf.random.uniform([1, 3], minval=0, maxval=64, dtype=tf.int64), dtype=tf.int64)
            res = tf.raw_ops.DenseToCSRSparseMatrix(
                dense_input=dense_input,
                indices=indices,
            )
    except:
        pass
```

### Relevant log output

```shell
Floating point exception (core dumped)
```
</details>

**Comment by tiruk007** [comment]
@sushreebarsa
I was able to reproduce the issue on Colab using TF v2.10 in CPU and GPU. Please find the gist [CPU](https://colab.research.google.com/gist/tiruk007/57a60509403e7007cffdceb06a164c93/58179cpu.ipynb) and [GPU](https://colab.research.google.com/gist/tiruk007/40a2caf3357324ee156816dd54c9ff5a/58179gpu.ipynb) for reference.
Thank you!

**Comment by sachinprasadhs** [comment]
tf.raw_ops are basically designed to be consumed by downstream library users, not end users.
Usually, these APIs are less strict in terms of validation etc., which is fine since mainly library writers are supposed to use these symbols.
If available, please prefer high level API for general use case scenarios.
Refer this [RFC](https://github.com/tensorflow/community/blob/master/rfcs/20181225-tf-raw-ops.md#design-proposal) for more details. Thanks!

**Comment by mihaimaruseac** [comment]
While these are internal APIs, the check failure could lead to denial of service. In the past this used to be considered a vulnerability, but since the impact is bounded and preventable, now the policy is for these to be files as issues and resolved whenever someone from community wants to pick them up (they're very good first issues). Please don't auto-close these!

**Comment by SuryanarayanaY** [comment]
Hi @shijy16 ,

The issue got fixed in tf-nightly)2.16.0-dev20231129). Now the Op is raising intended exception. Please refer to attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/9dead2bfeae174222e84cba3754e33d4/58179_gpu_nightly-2-16-dev.ipynb#scrollTo=kMLqLRgjFq-S).
